propagation.go

```diff
@@ -51,14 +51,20 @@ type Extractor interface {
 	Extract(carrier interface{}) (SpanContext, error)
 }
 
+// Propagator is a composite type embedding both an Injector and an Extractor.
+type Propagator interface {
+	Injector
+	Extractor
+}
+
 type textMapPropagator struct {
 	headerKeys  *HeadersConfig
 	metrics     Metrics
 	encodeValue func(string) string
 	decodeValue func(string) string
 }
 
-func newTextMapPropagator(headerKeys *HeadersConfig, metrics Metrics) *textMapPropagator {
+func NewTextMapPropagator(headerKeys *HeadersConfig, metrics Metrics) *textMapPropagator {
```

Comment:
exported func NewTextMapPropagator returns unexported type *jaeger.textMapPropagator, which can be annoying to use